src/core.js

```diff
@@ -239,9 +240,11 @@ jQuery.extend( {
 			return false;
 		}
 
-		// If the function hasn't returned already, we're confident that
-		// |obj| is a plain object, created by {} or constructed with new Object
-		return true;
+		// Own properties are enumerated firstly, so to speed up,
+		// if last one is own, then all properties are own
+		for ( key in obj ) {}
+
+		return key === undefined || hasOwn.call( obj, key );
```

Comment:
We can simplify it later on master, for 2.2-stable it'd be good to land quickly & release as users of our AMD modules are waiting for a critical fix.